Wizardry/Core/StatScreen/DrawPages/DrawPage1.c

```diff
@@ -118,115 +118,112 @@ STATIC_DECLAR void DrawStatWithBarRework(int num, int x, int y, int base, int to
 extern u8 gStatScreenStatsGrowthsToggle; 
 inline STATIC_DECLAR int DisplayGrowths(void) 
 { 
-	return gStatScreenStatsGrowthsToggle;
+    return gStatScreenStatsGrowthsToggle;
 } 
 
 STATIC_DECLAR int GetGrowthBonusOffset(int diff) 
 { 
-	diff = ABS(diff); 
-	int result = 1; 
-	if (diff > 99) 
+    diff = ABS(diff); 
+    int result = 1; 
+    if (diff > 99) 
     { 
         result++; 
     } 
-	return result; 
+    return result; 
 } 
 
 STATIC_DECLAR void DrawGrowthWithDifference(int x, int y, int base, int modified)
 {
     int diff = modified - base;
-	int offset = 0; 
-	if (base > 99) 
+    int offset = 0; 
+    if (base > 99) 
     { 
         offset++; 
     } 
-	if ((base < 100) && (diff < 100)) 
+    if ((base < 100) && (diff < 100)) 
     { 
         offset++; 
     } 
     PutNumberOrBlank(gUiTmScratchA + TILEMAP_INDEX(x + offset, y), TEXT_COLOR_SYSTEM_BLUE, base);
-	offset += GetGrowthBonusOffset(diff); 
+    offset += GetGrowthBonusOffset(diff); 
     PutNumberBonus(diff, gUiTmScratchA + TILEMAP_INDEX(x + offset, y));
 }
 
 struct TalkEventCond 
 {
     u16 eventType;
     u16 flag;
-	u32 eventPointer;
+    u32 eventPointer;
     u8 pidA;
     u8 pidB;
     u16 fillerA;
-	#ifndef FE6 
+    #ifndef FE6 
     u16 unkC;
     u16 unkE;
-	#endif 
+    #endif 
 };
 
 STATIC_DECLAR int _GetTalkee(int unitID) 
 { 
-	const struct TalkEventCond* talkCond = GetChapterEventDataPointer(gPlaySt.chapterIndex)->characterBasedEvents;
-	int flag, pid; 
-	struct Unit* unit; 
-	for (int i = 0; i < 255; ++i) 
-	{ 
-		if (!talkCond[i].eventType) 
-		{ 
-			break; 
-		} 
-		if (talkCond[i].pidA != unitID) 
-		{ 
-			continue; 
-		} 
-		flag = talkCond[i].flag; 
-		if (flag) 
-		{ 
-			if (CheckFlag(flag)) 
-			{ 
-				continue; 
-			} 
-		} 
-		pid = talkCond[i].pidB; 
-		#ifdef FE7 
-		if (talkCond[i].unkC) // I don't know if this does anything in FE8 
-		{ 
-			if (gMode != talkCond[i].unkC) continue; // any / Eliwood / Hector mode 
-		} 
-		#endif 
-		if (pid) 
-		{ 
-			//unit = GetUnitStructFromEventParameter(pid); 
-			unit = GetUnitFromCharId(pid); 
-			if (!UNIT_IS_VALID(unit)) continue; 
-			
-			if (unit->state & (US_DEAD|US_NOT_DEPLOYED|US_BIT16)) continue; 
-			
-			return pid;
-		} 
-	} 
-	return 0; 
+    const struct TalkEventCond* talkCond = GetChapterEventDataPointer(gPlaySt.chapterIndex)->characterBasedEvents;
+    int flag, pid; 
+    struct Unit* unit; 
+    for (int i = 0; i < 255; ++i) 
+    { 
+        if (!talkCond[i].eventType) 
+        { 
+            break; 
+        } 
+        if (talkCond[i].pidA != unitID) 
+        { 
+            continue; 
+        } 
+        flag = talkCond[i].flag; 
+        if (flag) 
+        { 
+            if (CheckFlag(flag)) 
+            { 
+                continue; 
+            } 
+        } 
+        pid = talkCond[i].pidB; 
+        #ifdef FE7 
```

Comment:
apparently this is not FE7

Reply:
I use this section of code in the self randomizers, and fe7 limits talk events based on eliwood / hector mode.